tests/PropertyAccess/ReflectionPropertyAccessorTest.php

```diff
@@ -0,0 +1,260 @@
+
+        $this->assertEquals($expected, $object);
+    }
+
+    public function testSetValueWithTheDecoratedAccessorWhenPossible()
+    {
+        $object = new DummyWithPublicProperty();
+        $property = 'val';
+        $value = 'bar';
+
+        $expected = new DummyWithPublicProperty();
+        $expected->val = 'bar';
+
+        $decoratedAccessorProphecy = $this->prophesize(PropertyAccessorInterface::class);
+        $decoratedAccessorProphecy
+            ->setValue($object, $property, $value)
+            ->will(
+                function ($args) {
+                    $args[0]->{$args[1]} = $args[2];
+                }
+            )
+        ;
+        /** @var PropertyAccessorInterface $decoratedAccessor */
+        $decoratedAccessor = $decoratedAccessorProphecy->reveal();
+
+        $accessor = new ReflectionPropertyAccessor($decoratedAccessor);
+        $accessor->setValue($object, $property, $value);
+
+        $this->assertEquals($expected, $object);
+
+        $decoratedAccessorProphecy->setValue(Argument::cetera())->shouldHaveBeenCalledTimes(1);
+    }
+
+    public function testGetPrivateValueOnNoSuchPropertyException()
+    {
+        $property = 'val';
+        $value = 'foo';
+        $object = new DummyWithPrivateProperty($value);
+
+        $decoratedAccessorProphecy = $this->prophesize(PropertyAccessorInterface::class);
+        $decoratedAccessorProphecy
+            ->getValue($object, $property)
+            ->willThrow(NoSuchPropertyException::class)
+        ;
+
+        /** @var PropertyAccessorInterface $decoratedAccessor */
+        $decoratedAccessor = $decoratedAccessorProphecy->reveal();
+
+        $accessor = new ReflectionPropertyAccessor($decoratedAccessor);
+        $actual = $accessor->getValue($object, $property);
+
+        $this->assertEquals($value, $actual);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException
+     * @expectedExceptionMessage Cannot read property "foo".
+     */
+    public function testThrowsAnOriginalExceptionIfPropertyDoesNotExist()
+    {
+        $property = 'foo';
+        $object = new DummyWithPrivateProperty();
+
+        $decoratedAccessorProphecy = $this->prophesize(PropertyAccessorInterface::class);
+        $decoratedAccessorProphecy
+            ->getValue($object, $property)
+            ->willThrow(new NoSuchPropertyException("Cannot read property \"$property\"."))
+        ;
+
+        /** @var PropertyAccessorInterface $decoratedAccessor */
+        $decoratedAccessor = $decoratedAccessorProphecy->reveal();
+
+        $accessor = new ReflectionPropertyAccessor($decoratedAccessor);
+
+        $accessor->getValue($object, $property);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException
+     * @expectedExceptionMessage Cannot read property "staticVal".
+     */
+    public function testThrowsAnOriginalExceptionIfPropertyIsStatic()
```

Comment:
I don't think we should support it (use cases look a bit edgy to me), but you set the rules. 😄 

Actually, I'm not even sure the property accessor component is meant to be used to access any static property.